lib/jxl/enc_fast_lossless.cc

```diff
@@ -3471,10 +3480,38 @@ void PrepareDCGlobalPalette(bool is_single_group, size_t width, size_t height,
   }
 }
 
+template <size_t nb_chans>
+bool detect_palette(const unsigned char* r, size_t width,
+                    std::vector<uint32_t>& palette) {
+  size_t x = 0;
+  bool collided = false;
+  // this is just an unrolling of the next loop
+  for (; x + 7 < width; x += 8) {
+    uint32_t p[8] = {}, index[8];
+    for (int i = 0; i < 8; i++) memcpy(&p[i], r + (x + i) * nb_chans, 4);
+    if (nb_chans < 4)
```

Comment:
This is probably unnecessary (the compiler will likely skip a & 0xFFFFFFFF on a 32-bit integer), but if you leave it, please add braces :)

Reply:
Right, it's unnecessary.